pkg/agent/agent.go

```diff
@@ -1132,23 +1132,19 @@ func getRoundInfo(bridgeClient ovsconfig.OVSBridgeClient) types.RoundInfo {
 	return roundInfo
 }
 
-func (i *Initializer) getNodeMTU(transportInterface *net.Interface) (int, error) {
+func (i *Initializer) getInterfaceMTU(transportInterface *net.Interface) (int, error) {
 	if i.mtu != 0 {
 		return i.mtu, nil
 	}
 	mtu := transportInterface.MTU
-	// Make sure mtu is set on the interface.
+	// Make sure MTU is set on the interface.
 	if mtu <= 0 {
 		return 0, fmt.Errorf("Failed to fetch Node MTU : %v", mtu)
 	}
+
+	mtu -= i.networkConfig.MTUDeduction
+
 	if i.networkConfig.TrafficEncapMode.SupportsEncap() {
-		if i.networkConfig.TunnelType == ovsconfig.VXLANTunnel {
-			mtu -= config.VXLANOverhead
-		} else if i.networkConfig.TunnelType == ovsconfig.GeneveTunnel {
-			mtu -= config.GeneveOverhead
-		} else if i.networkConfig.TunnelType == ovsconfig.GRETunnel {
-			mtu -= config.GREOverhead
-		}
 		if i.nodeConfig.NodeIPv6Addr != nil {
 			mtu -= config.IPv6ExtraOverhead
 		}
```

Comment:
Could you remind me how we calculate MTU when Wireguard (for intra-cluster traffic) is enabled? I only see IPsec is considered in this func.

Reply:
Here is the way to calculate MTU: https://github.com/antrea-io/antrea/blob/main/pkg/agent/agent.go#L1060-L1070

Reply:
WireGuard does not impact Pod MTU?

Reply:
I think it doesn't impact Pod MTU since the traffic is sent out of antrea tunnel interface, then being routed to WireGuard interface, @hjiajing @xliuxu may have a better explanation, could you double check?  thanks.

Reply:
I agree with Lan. The traffic is be forwarded to the host network interface, so I think it will not impact the Pod MTU.